src/cloudant/database.py

```diff
@@ -792,6 +793,128 @@ def shards(self):
+                    data.get('name'),
+                    **data.get('def', {})
+                ))
+            elif data.get('type') == 'text':
+                indexes.append(SearchIndex(
+                    self,
+                    data.get('ddoc'),
+                    data.get('name'),
+                    **data.get('def', {})
+                ))
+            elif data.get('type') == 'special':
+                indexes.append(SpecialIndex(self, **data.get('def', {})))
+            else:
+                raise CloudantException('Unexpected index content: {0} found.')
+        return indexes
+
+    def create_index(
+            self,
+            design_document_id=None,
+            index_name=None,
+            index_type='json',
+            **kwargs
+        ):
+        """
+        Creates either a JSON or a text index in the remote database.
+
+        :param str index_type: The type of the index to create.  Can
+            be either 'text' or 'json'.  Defaults to 'json'.
+        :param str design_document_id: Optional identifier of the design
+            document in which the index will be created. If omitted the default
+            is that each index will be created in its own design document.
+            Indexes can be grouped into design documents for efficiency.
+            However, a change to one index in a design document will invalidate
+            all other indexes in the same document.
+        :param str index_name: Optional name of the index. If omitted, a name
+            will be generated automatically.
+        :param list fields: A list of fields that should be indexed.  For JSON
+            indexes, the fields parameter is mandatory and should follow the
+            'sort syntax'.  For example ``fields=['name', {'age': 'desc'}]``
+            will create an index on the 'name' field in ascending order and the
+            'age' field in descending order.  For text indexes, the fields
+            parameter is optional.  If it is included then each field element
+            in the fields list must be a single element dictionary where the
+            key is the field name and the value is the field type.  For example
+            ``fields=[{'name': 'string'}, {'age': 'number'}]``.  Valid field
+            types are ``'string'``, ``'number'``, and ``'boolean'``.
+        :param dict default_field: Optional parameter that specifies how the
+        ``$text`` operator can be used with the index.  Only valid when
+            creating a text index.
+        :param dict selector: Optional parameter that can be used to limit the
+            index to a specific set of documents that match a query. It uses
+            the same syntax used for selectors in queries.  Only valid when
+            creating a text index.
+
+        :returns: An Index object representing the index created in the
+            remote database
+        """
+        index = None
+        if index_type == 'json':
```

Comment:
Added an index_constants module in 21c9d5dd5d95925c23a67fe9318f26c8a10e07b3 and changed all references in the database module to reference the index constants in 5740aa251ca744ad62d53bb36986bf68e2da4340.